vector/src/main/java/im/vector/app/features/location/LocationSharingFragment.kt

```diff
@@ -212,30 +195,16 @@ class LocationSharingFragment @Inject constructor(
     }
 
     private val foregroundLocationResultLauncher = registerForPermissionsResult { allGranted, deniedPermanently ->
-        if (allGranted && checkPermissions(PERMISSIONS_FOR_BACKGROUND_LOCATION_SHARING, requireActivity(), backgroundLocationResultLauncher)) {
-            startLiveLocationSharing()
-        } else if (deniedPermanently) {
-            handleMissingBackgroundLocationPermission()
-        }
-    }
-
-    private val backgroundLocationResultLauncher = registerForPermissionsResult { allGranted, deniedPermanently ->
         if (allGranted) {
             startLiveLocationSharing()
         } else if (deniedPermanently) {
-            handleMissingBackgroundLocationPermission()
+            activity?.onPermissionDeniedDialog(R.string.denied_permission_generic)
         }
     }
 
     private fun tryStartLiveLocationSharing() {
         // we need to re-check foreground location to be sure it has not changed after landing on this screen
-        if (checkPermissions(PERMISSIONS_FOR_FOREGROUND_LOCATION_SHARING, requireActivity(), foregroundLocationResultLauncher) &&
-                checkPermissions(
-                        PERMISSIONS_FOR_BACKGROUND_LOCATION_SHARING,
-                        requireActivity(),
-                        backgroundLocationResultLauncher,
-                        R.string.location_in_background_missing_permission_dialog_content
-                )) {
+        if (checkPermissions(PERMISSIONS_FOR_FOREGROUND_LOCATION_SHARING, requireActivity(), foregroundLocationResultLauncher)) {
```

Comment:
The restriction on location permission access in background has been added in Android 11 so it is supposed to support these versions. But I will check it to confirm.

Reply:
I have done some testing with Android 12, 11, 10, 9 and had no issues. I cannot test with Android 5 since the emulator does not run on my laptop with this version.